sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -220,6 +235,12 @@ private static void killUnneededTasks(StateStore stateStore, Set<String> taskToD
 
     @Override
     protected OfferResponse processOffers(Collection<Protos.Offer> offers, Collection<Step> steps) {
+        if (serviceSpec.getGoal() == GoalState.FINISH
```

Comment:
This is just slightly hacky.  We call `processOffers` regardless of whether we have any offers to process so it'll definitely work.  It's our core polling mechanism I suppose.  Ideally I think we'd check for eligibility for uninstall independently of our offer processing cycle.  I think changing this as part of this PR would only make it bigger, but it would be good to file a blocking JIRA to revisit our design around service lifecycle vs. offer processing.

Reply:
There's currently a TODO tucked away in MesosEventClient (merged previously), proposing a separate getStatus() (or similar) call for misc checks like this: https://github.com/mesosphere/dcos-commons/blob/master/sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/MesosEventClient.java#L76

Filed INFINITY-3493 to track it as well.